### PR TITLE
RichardsMechanics restart

### PR DESCRIPTION
**Pull request by endJunction**
Adds restart possibility to the RichardsMechanics process.

Several parts are copies from SD or HM. Some improvements too, especially on the `getSigma` and `setSigma` and the likes.

Commit-wise review possible.

1. [ ] Feature description was added to the [changelog](https://github.com/ufz/ogs/wiki/Release-notes-6.3.1)
2. [x] Tests covering your feature were added?
3. [x] Any new feature or behavior change was documented?

### CODE REVIEW

**Comment by endJunction** [comment]
@FZill You could port the last three commits to HM, if you want: "Store darcy velocity", "Generalize a transposition", and "Replace setSigma/setPorosity".

File: ProcessLib/RichardsMechanics/RichardsMechanicsFEM-impl.h [hunk 1]
```diff
@@ -824,6 +878,67 @@ void RichardsMechanicsLocalAssembler<ShapeFunctionDisplacement,
...
 }
 
+template <typename ShapeFunctionDisplacement, typename ShapeFunctionPressure,
+          typename IntegrationMethod, int DisplacementDim>
+std::size_t RichardsMechanicsLocalAssembler<
+    ShapeFunctionDisplacement, ShapeFunctionPressure, IntegrationMethod,
+    DisplacementDim>::setKelvinVector(double const* values,
+                                      KelvinVectorType IpData::*member)
+{
+    auto const kelvin_vector_size =
```
Comment (TomFischer) [inline ProcessLib/RichardsMechanics/RichardsMechanicsFEM-impl.h:888]: In `getSigma()` the `kelvin_vector_size` is a 
```constexpr int kelvin_vector_size = MathLib::KelvinVector::KelvinVectorDimensions<DisplacementDim>::value;```
Why here a different implementitation?

File: ProcessLib/RichardsMechanics/RichardsMechanicsFEM-impl.h [hunk 2]
```diff
@@ -824,6 +878,67 @@ void RichardsMechanicsLocalAssembler<ShapeFunctionDisplacement,
...
+        MathLib::KelvinVector::KelvinVectorDimensions<DisplacementDim>::value;
+    auto const n_integration_points = _ip_data.size();
+
+    auto matrix_mapped_values =
```
Comment (TomFischer) [inline ProcessLib/RichardsMechanics/RichardsMechanicsFEM-impl.h:892]: I wonder if `matrix_mapped_values` could be `const`? What happens in 
```C++
MathLib::KelvinVector::symmetricTensorToKelvinVector(
                matrix_mapped_values.col(ip));
```
?

Comment (endJunction) [reply ProcessLib/RichardsMechanics/RichardsMechanicsFEM-impl.h]: const added. Thanks.

**Pull request merged**